frontend/javascripts/admin/tasktype/task_type_create_view.tsx

```diff
@@ -43,464 +62,448 @@ function isValidMagnification(_rule, value) {
-    };
-    // @ts-expect-error ts-migrate(2339) FIXME: Property 'resolutionRestrictions' does not exist o... Remove this comment to see the full error message
-    delete formValues.settings.resolutionRestrictions;
-    const form = this.formRef.current;
-
-    if (!form) {
-      Toast.info(messages["ui.no_form_active"]);
-      return;
-    }
-
-    form.setFieldsValue(formValues);
-
-    if (taskType?.recommendedConfiguration != null) {
-      // Only "activate" the recommended configuration checkbox if the existing task type contained one
-      this.setState({
-        useRecommendedConfiguration: true,
-      });
+    const defaultFormValues: Partial<FormValues> = _.merge({}, defaultValues, taskType);
+    form.setFieldsValue(defaultFormValues);
+
+    if (taskType?.recommendedConfiguration) {
+      //  Only "activate" the recommended configuration checkbox if the existing task type contained one
+      setUseRecommendedConfiguration(true);
+      form.setFieldValue(
+        ["recommendedConfiguration"],
+        jsonStringify(taskType?.recommendedConfiguration),
+      );
     }
-  }
 
-  async fetchData() {
-    this.setState({
-      isFetchingData: true,
-    });
-    const editableTeams = await getEditableTeams();
-    this.setState({
-      teams: editableTeams,
-      isFetchingData: false,
-    });
+    if (taskType?.settings.resolutionRestrictions)
+      form.setFieldValue(["isResolutionRestricted"], true);
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'formValues' implicitly has an 'any' typ... Remove this comment to see the full error message
-  onFinish = async (formValues) => {
-    if (!this.state.useRecommendedConfiguration) {
-      formValues.recommendedConfiguration = null;
-    }
-
-    const { recommendedConfiguration, settings: rawSettings, ...rest } = formValues;
-    const settings = getMagnificationAdaptedSettings(rawSettings);
+  async function onFinish(formValues: FormValues) {
+    const {
+      settings,
+      teamId,
+      recommendedConfiguration,
+      isResolutionRestricted: _isResolutionRestricted,
+      ...rest
+    } = formValues;
+    const teamName = teams.find((team) => team.id === teamId)!["name"];
 
     if (!settings) {
       return;
     }
 
+    // FormItems which are not rendered/hidden are not serialized by onFinish
+    // add them manually
+    if (!settings.resolutionRestrictions) {
+      settings.resolutionRestrictions = { min: undefined, max: undefined };
+    }
+
     const newTaskType = {
       ...rest,
+      teamId,
       settings,
-      recommendedConfiguration:
-        recommendedConfiguration != null ? JSON.parse(recommendedConfiguration) : null,
+      recommendedConfiguration: JSON.parse(recommendedConfiguration),
     };
 
-    if (this.props.taskTypeId) {
-      await updateTaskType(this.props.taskTypeId, newTaskType);
+    if (taskTypeId) {
+      const updatedTaskType = _.omit({ id: taskTypeId, teamName, ...newTaskType }, [
```

Comment:
When updating a task type that has a resolution restriction the server complains if you resend it even if it has not changed. So we need to remove it again :-/